### DISCUSSION TITLE
openDTU Access - openDTU Zugriff

### DISCUSSION THREAD

**Discussion by Klausimodo** [tbnobody/OpenDTU#854] [Q&A]
I am currently working on a foreign openDTU to change the pin mapping.
To do this, I took the openDTU home with me. I know all the passwords. The DTU runs in a network 192.168.5.x. My home network works with 192.168.8.x. So a connection is not possible, especially as the DTU does not know my router data.
So I wanted to enable a connection via AP in the DTU. Shortly after switching on, I see the DTU. When I try to connect to it, I always get the answer that the password is wrong. openDTU42 as the password does not work. All other passwords that might come into question do not work either. If I enter the password again, it says "no connection". So I disconnect the DTU from the power supply and switch it on again. Then I try the same thing with the next possible password. The effect is the same. The first time it says the password is wrong. The second time it says "no connection".
I don't know how to get on without flashing again.

Und in deutsch:
Ich arbeite gerade an einer fremden openDTU, um das Pin-Mapping zu ändern.
Dazu habe ich die openDTU mit nach Hause genommen. Ich kenne alle Passwörter. Die DTU läuft in einem Netz 192.168.5.x. Mein Heim-Netz arbeitet mit 192.168.8.x. Also ist eine Verbindung so nicht möglich, zumal die DTU ja meine Routerdaten nicht kennt.
Also wollte ich eine Verbindung via AP in der DTU ermöglichen. Kurz nach dem Einschalten sehe ich die DTU. Wenn ich mich mit ihr verbinden will, bekomme ich jedesmals die Antwort, dass das Passwort falsch sei. openDTU42 als Passwort funktioniert nicht. Auch alle anderen Passwörter, die vielleicht in Frage kommen könnten, funktionieren nicht. Gebe ich das Passwort erneut ein, heißt es "keine Verbindung". Also mache ich die DTU stromlos und schalte sie wieder ein. Danach versuche ich es genau so mit dem nächsten, möglichen Passwort. Der Effekt ist derselbe. Beim ersten Mal heisst es, das Passwort sei falsch. Beim zweiten Mal "keine Verbindung".
Ich weiß nicht, wie ich - ohne erneut zu flashen - weiterkommen soll.

**Comment by DPO99** [comment]
Hab das mal nachgestellt. Versuch mal das Admin Passwort der Oberfläche. Das hat bei mir funktioniert. Das Problem ist, dass der AP schnell ausgeht (glaube 3 min.) wenn Du jedesmal erst 5 Passwörter vor dem eigentlichen versuchst, ist er ggf. schon wieder aus. Wie gesagt Admin Passwort hat umgehend geklappt.

**Reply by Klausimodo** [reply]
Genau das habe ich versucht. Ich weiß, dass man nur wenig Zeit hat und ich kenne alle in Frage kommenden Passwörter. Es hat nicht geklappt.
Beim genaueren Lesen stelle ich sogar fest, dass diese 3 Minuten-Frist erst dann eintritt, wenn das Teil sich mit einem WiFi-AP verbunden hat.

**Comment by DcFamas** [comment]
Das Passwort lautet nicht mehr openDTU42. Siehe hier;

https://github.com/tbnobody/OpenDTU#first-configuration

"Navigate to Settings --> Network Settings and enter your WiFi credentials. **The username to access the config menu is "admin" and the password the same as for accessing the Access Point** (default: "openDTU42")."

Ergo; wenn er das Kennwort für den admin-Zugriff geändert hat, ist das nun das Kennwort des APs.

**Reply by Klausimodo** [reply]
Wenn es so wäre, wäre das ja fein. 
Da steht aber nur, dass in der Anfangskonfiguration beide Passwörter gleich sind. Da steht nicht, dass mit Änderung des admin-Passworts auch das AP-Passwort geändert wird. Und weil ich das Admin-Passwort kenne und das aber beim Anmelden im AP nicht funktioniert, bin ich eben so irritiert.

**Reply by DcFamas** [reply]
Okay, davon bin ich leider ausgegangen. Gegebenfalls einfach neu flashen?

**Reply by DPO99** [reply]
Hast Du das Admin Passwort selbst einmal getestet, als das Ding noch im Netzwerk war oder verlässt Du dich hier auf eine Aussage? 

Ansonsten hast du 3 Optionen:
-Ding zurück ins originale Netzwerk und Admin Passwort verifizieren/testen
-Neuste Firmware OTA flashen (im alten Netz) und dann das gleiche nochmal testen
-Neu flashen (ggf. mal mit Backup testen aber vielleicht nimmst den Fehler dann mit).

**Comment by tbnobody** [comment]
Nach dem einschalten sucht die DTU 15 Sekunden lang nach einem bekannten Wlan. Wenn keines gefunden wird, wird 600 Sekunden pausiert und wieder 15 Sekunden gesucht. 

Das ist so "seltsam" implementiert, weil während dieser 15 Sekunden wechselt der WiFi Chip durch alle möglichen Kanäle durch. Es gibt jedoch für AP und Station Modus nur einen RF Chip. Sprich auch der AP Kanal ändert hier den Kanal. Es ist daher in diesen 15 Sekunden schwierig sich mit dem AP zu verbinden. Daher wird das Suchen pausiert um einen stabileren AP-Modus zu erhalten.

Mein Ansatz wäre, den Verbindungsversuch erst 30 Sekunden nach dem Einschalten zu starten. 


> Da steht aber nur, dass in der Anfangskonfiguration beide Passwörter gleich sind.

Ich finde schon, dass in dem Dialog recht eindeutig steht, dass das Admin Passwort für beides verwendet wird.
![image](https://user-images.githubusercontent.com/1753222/235112298-ba29f32a-bed3-40db-9cd8-40be97477803.png)

**Reply by Klausimodo** [reply]
Asche auf mein Haupt! Natürlich steht genau das in Sachen Passwort dort! An diese Stelle kam ich bis eben aber nicht. Ich hatte mich auf eine andere Formulierung auf der Webseite https://github.com/tbnobody/OpenDTU#first-configuration gestützt und zwar "Navigate to Settings --> Network Settings and enter your WiFi credentials. The username to access the config menu is "admin" and the password the same as for accessing the Access Point (default: "openDTU42")." Und daraus ging für mich nicht hervor, dass das so ist, wie du beschrieben hast. 

Entweder habe ich eben das offene Zeitfenster gefunden - ich kam nämlich jetzt mit dem Admin-Passwort rein - oder es gibt eine weitere Ursache. Da das ja zig Mal nicht klappte und es zuletzt nur zu wackeligen Verbindungen kam, die allenfalls eine halbe Minute hielt, habe ich die DTU etwas mehr von der WLAN-Antenne des PC entfernt. Die Entfernung betrug vorher ca. 50cm. Ob das zu der nun haltenden Verbindung beigetragen hat, ist aber reine Spekulation.
Nun geht die Geschichte weiter. Es ging ja ursprünglich um das PIN-Mapping. Im Neuen sind 2 PINS anders belegt.
[
    {
        "name": "ESP32-WROOM-32 30 pin",
        "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 27,
            "en": 33,
            "cs": 5
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": -1,
            "clk_mode": -1
        }
    }
]
Das habe ich nun hochladen können und es wurde auch übernommen. Aber ein Miteffekt war, dass das Passwort nun wieder in openDTU42 geändert ist. Bin schon wieder irritiert.

**Comment by tbnobody** [comment]
> Aber ein Miteffekt war, dass das Passwort nun wieder in openDTU42 geändert ist.

Das ist durch ein Upload der pin_mapping.json definitv nicht der Fall. 
Bist du sicher, dass du hier
![image](https://user-images.githubusercontent.com/1753222/235125586-1cf3e90d-61f7-453e-bfbd-230b3bbbcab2.png)
statt "Main Config" die "Pin Mapping" ausgewählt hast? Ansonsten hast du natürlich deine Main Config mit der PinMapping überschrieben. Was erstmal keinen Sinn macht und dadurch die "Werkseinstellungen" angewendet werden.

**Reply by Klausimodo** [reply]
Genau das war offensichtlich der Fehler. Danke für den Tipp und die Geduld mit einem Anfänger!
